velox/functions/sparksql/String.h

```diff
@@ -211,4 +211,147 @@ struct EndsWithFunction {
   }
 };
 
+/// trim functions
+/// ltrim(srcStr) -> varchar
+///     Removes leading 0x20(space) characters from srcStr.
+/// ltrim(trimStr, srcStr) -> varchar
+///     Remove leading specified characters from srcStr. The specified character
+///     is any character contained in trimStr.
+/// rtrim(srcStr) -> varchar
+///     Removes trailing 0x20(space) characters from srcStr.
+/// rtrim(trimStr, srcStr) -> varchar
+///     Remove trailing specified characters from srcStr. The specified
+///     character is any character contained in trimStr.
+/// trim(srcStr) -> varchar
+///     Remove leading and trailing 0x20(space) characters from srcStr.
+/// trim(trimStr, srcStr) -> varchar
+///     Remove leading and trailing specified characters from srcStr. The
+///     specified character is any character contained in trimStr.
+template <typename T, bool leftTrim, bool rightTrim>
+struct TrimFunctionBase {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // Results refer to strings in the first argument.
```

Comment:
This is true for single-arg functions, but 2-arg functions reuse results from the second argument.